allensdk/brain_observatory/ecephys/ecephys_session.py

```diff
@@ -1180,26 +1181,18 @@ def _build_stimulus_presentations(
             stimulus_presentations, default_column_renames, ignore_case=False
         )
 
-        # pandas groupby ops ignore nans, so we need a new "nonapplicable"
-        # value that pandas does not recognize as null ...
-        # stimulus_presentations.replace("", nonapplicable, inplace=True)
-
-        # pandas does not automatically convert boolean cols for fillna
-        # boolean_colnames = stimulus_presentations.dtypes[
-        #     stimulus_presentations.dtypes == "boolean"].index
-        # col_type_map = {colname: "object" for colname in boolean_colnames}
-        # stimulus_presentations = stimulus_presentations.astype(
-        #     col_type_map).fillna(nonapplicable)
+        stimulus_presentations.replace("", nonapplicable, inplace=True)
 
         # eval str(numeric) and str(lists), convert lists to tuple for
         # dict key compatibility
-        exclude_columns = ["stimulus_name"]
-        for colname in stimulus_presentations.columns:
-            if colname not in exclude_columns:
-                stimulus_presentations[colname] = stimulus_presentations[
-                    colname
-                ].apply(naming_utilities.eval_str)
-
+        
+        col_list = ["phase, size, spatial_frequency"]
```

Comment:
I changed the logic here to eval/tuple by specifying columns instead of creating rules.